scripts/common.Makefile

```diff
@@ -124,7 +124,8 @@ pylint: $(REPO_BASE_DIR)/.pylintrc ## runs pylint (python linter) on src and tes
 	@pylint --rcfile="$(REPO_BASE_DIR)/.pylintrc" -v $(CURDIR)/src $(CURDIR)/tests
 
 
-.PHONY: mypy
+.PHONY: mypy typecheck
+typecheck: mypy
```

Comment:
So from what I now understand we have common command across all repos to run `make typecheck`.
Currently if you run `make codestyle` as a last step it runs mypy. Will `make codestyle` run the same check of `make typecheck` (via the docker image). If yes, that is great!

Reply:
I propose to deprecate ``make codestyle`` and replace by ``make typecheck``because:
1. ``codestyle`` does isort, black (already done with precommit) and then pylint (already in tests and CI job) and then mypy
2. mypy is about type checking and not code styling
3. it is not uniform repo-wide

Reply:
> 1. `codestyle` does isort, black (already done with precommit) and then pylint (already in tests and CI job) and then mypy

 For example in my setup, I find `codestyle` very useful since I do not run precommit. 

> 2. mypy is about type checking and not code styling

 I can also go with removing the mypy from the codestyle 

I'd go with the above changes if possible.

Reply:
@GitHK there are new functions in the ``makefile`` that can run all/parts of the pre-commit (let's talk about it)

You can also create a "convenience" recipe in the local package/service ``Makefile`` with whatever you like